test/v5/kyberTradeLogic.js

```diff
@@ -1,436 +1,549 @@
+                srcQty = new BN(1000).mul(new BN(10).pow(srcDecimals));
+                fees = [takerFeeBps, platformFeeBps];
+            });
 
-async function assertSameEtherBalance(accountAddress, expectedBalance) {
-    let balance = await Helper.getBalancePromise(accountAddress);
-    Helper.assertEqual(balance, expectedBalance, "wrong ether balance");
-}
+            it("T2E, no hint", async() => {
+                reserveCandidates = await fetchReservesRatesFromTradeLogic(tradeLogic, reserveInstances, srcToken.address, srcQty, takerFeeBps, true);
+                bestReserve = await nwHelper.getBestReserveAndRate(reserveCandidates, srcToken.address, ethAddress, srcQty, takerFeeBps);
+                result = await tradeLogic.calcRatesAndAmounts(srcToken.address, ethAddress, srcQty, fees, emptyHint);
+            });
 
-async function assertSameTokenBalance(accountAddress, token, expectedBalance) {
-    let balance = await token.balanceOf(accountAddress);
-    Helper.assertEqual(balance, expectedBalance, "wrong token balance");
-}
+            it("E2T, no hint", async() => {
 
-//returns random integer between min (inclusive) and max (inclusive)
-function getRandomInt(min, max) {
-    min = Math.ceil(min);
-    max = Math.floor(max);
-    return Math.floor(Math.random() * (max - min + 1)) + min;
-}
+            });
 
-function getBestReserve(rateArr, reserveArr) {
-    bestReserve = {
-        address: '',
-        rateNoFee: new BN(0), 
-        rateWithNetworkFee: new BN(0)
-    }
+            it("T2T, no hint", async() => {
 
-    for (let i=0; i < rateArr.length; i++) {
-        let rate = rateArr[i];
-        let rateForComparison = rate;
-        if (isFeePaying[i]) {
-            rateForComparison = rate.mul(BPS.sub(takerFeesBps)).div(BPS);
-        }
-
-        if (rateForComparison.gt(bestReserve.rateNoFee)) {
-            bestReserve.rateNoFee = rate;
-            bestReserve.rateWithNetworkFee = rateForComparison;
-            if (reserveArr.length) {
-                bestReserve.address = reserveArr[i].address;
-            }
-        }
-    }
-    return bestReserve;
-}
+            });
 
-function log(str) {
-    console.log(str);
+            it("T2E, mask in hint", async() => {
+
+            });
+
+            it("E2T, mask in hint", async() => {
+
+            });
+
+            it("T2T, mask in hint", async() => {
+
+            });
+
+            it("T2E, mask out hint", async() => {
+
+            });
+
+            it("E2T, mask out hint", async() => {
+
+            });
+
+            it("T2T, mask out hint", async() => {
+
+            });
+
+            it("T2E, split hint", async() => {
+
+            });
+
+            it("E2T, split hint", async() => {
+
+            });
+
+            it("T2T, split hint", async() => {
+
+            });
+        })
+    })
+});
+
+async function fetchReservesRatesFromTradeLogic(tradeLogicInstance, reserveInstances, tokenAddress, qty, takerFeeBps, isTokenToEth) {
```

Comment:
can't we use the global function in nw helper?

no logic is repeated.

I have one concern, if we find a bug in fetch function in nwHelper. and we fix it.
it will not be fixed here.
so can we combine them?